backend/src/api/game.rs

```diff
@@ -0,0 +1,59 @@
+use actix_web::error::{ErrorInternalServerError, ErrorNotFound};
+use actix_web::{post, web, HttpResponse, Responder};
+use hive_lib::game_type::GameType;
+use serde::{Deserialize, Serialize};
+
+use crate::model::challenge::GameChallenge;
+use crate::{db::util::DbPool, extractors::auth::AuthenticatedUser};
+
+#[derive(Deserialize)]
+pub struct NewGameRequest {
+    // Whether this challenge should be listed publicly
+    pub public: bool,
+
+    // Whether the game will be ranked
+    pub ranked: bool,
+
+    pub game_type: GameType,
+}
+
+#[derive(Serialize)]
+pub struct NewGameResponse {
+    challenge_url: String,
+}
+
+#[post("/game")]
+pub async fn create_game_challenge(
+    game: web::Json<NewGameRequest>,
+    auth_user: AuthenticatedUser,
+    pool: web::Data<DbPool>,
+) -> impl Responder {
+    GameChallenge::create(&auth_user.uid, &game, &pool)
+        .await
+        .map_err(|err| ErrorInternalServerError(format!("{:?}", err)))
+        .map(|challenge| {
+            let challenge_url = format!("/game/challenge/{}", challenge.id);
+            web::Json(NewGameResponse { challenge_url })
+        })
+}
+
+#[post("/game/challenge/{id}")]
+pub async fn accept_game_challenge(
+    id: web::Path<i32>,
+    auth_user: AuthenticatedUser,
+    pool: web::Data<DbPool>,
+) -> impl Responder {
```

Comment:
I think all the error handling should use our own errors, so this should return a `Result<_, ServerError>`